zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/ZNodeGroupAclUtil.java

```diff
@@ -33,4 +36,33 @@
   // Has the same effect as the ZK client using ZooDefs.Ids.CREATOR_ALL_ACL.
   public static final String SET_X509_CLIENT_ID_AS_ACL =
       ZNODE_GROUP_ACL_CONFIG_PREFIX + "setX509ClientIdAsAcl";
+  // A list of domain names that will have super user privilege, separated by ","
+  public static final String SUPER_USER_DOMAIN_NAME =
+      ZNODE_GROUP_ACL_CONFIG_PREFIX + "superUserDomainName";
+  // A list of znode path prefixes, separated by ","
+  // Znode whose path starts with the defined path prefix would have open read access
+  // Meaning the znode will have (world:anyone, r) ACL
+  public static final String OPEN_READ_ACCESS_PATH_PREFIX =
+      ZNODE_GROUP_ACL_CONFIG_PREFIX + "openReadAccessPathPrefix";
+
+  /**
+   * Get open read access path prefixes from config
+   * @return A set of path prefixes
+   */
+  public static Set<String> getOpenReadAccessPathPrefixes() {
+    String openReadAccessPathPrefixesStr = System.getProperty(OPEN_READ_ACCESS_PATH_PREFIX, "");
+    return Arrays.stream(openReadAccessPathPrefixesStr.split(","))
+        .filter(str -> str.length() > 0).collect(Collectors.toSet());
+  }
+
+  /**
+   * Get the domain names that are mapped to super user access privilege
+   * @return A set of domain names
+   */
+  public static Set<String> getSuperUserDomainNames() {
+    String superUserDomainNameStr =
+        System.getProperty(ZNodeGroupAclUtil.SUPER_USER_DOMAIN_NAME, "");
+    return Arrays.stream(superUserDomainNameStr.split(","))
+        .filter(str -> str.length() > 0).collect(Collectors.toSet());
```

Comment:
Commenting on this again - I think we should return an immutable constant (empty collection) when there are no elements or if the property is not found.

Reply:
I could do this ``` return Arrays.stream(superUserDomainNameStr.split(",")).filter(str -> str.length() > 0)
        .collect(Collectors.collectingAndThen(Collectors.toSet(), Collections::unmodifiableSet));``` to return an immutable set. But I don't think that's what you want. I think there is a tradeoff here, do we want to do one more check in order to prevent creating an empty set, or we could save this check for all the cases but allow an empty set in the cases where there are no elements. We can discuss offline on why you think it is critical.

Reply:
Synced up offline. Updated the code for efficient memory usage.